vortex-duckdb/src/convert/array/mod.rs

```diff
@@ -1,40 +1,73 @@
 mod data_chunk_adaptor;
 
-use duckdb::core::DataChunkHandle;
+use arrow_array::ArrayRef as ArrowArrayRef;
+use duckdb::core::{DataChunkHandle, SelectionVector};
 use duckdb::vtab::arrow::{
     WritableVector, flat_vector_to_arrow_array, write_arrow_array_to_vector,
 };
 use vortex_array::arrays::StructArray;
 use vortex_array::arrow::{FromArrowArray, IntoArrowArray};
+use vortex_array::compute::try_cast;
 use vortex_array::validity::Validity;
-use vortex_array::{Array, ArrayRef};
-use vortex_error::{VortexResult, vortex_err};
+use vortex_array::vtable::EncodingVTable;
+use vortex_array::{Array, ArrayRef, ArrayStatistics, ToCanonical};
+use vortex_dict::{DictArray, DictEncoding};
+use vortex_dtype::DType;
+use vortex_dtype::Nullability::NonNullable;
+use vortex_dtype::PType::U32;
+use vortex_error::{VortexExpect, VortexResult, vortex_err};
 
 use crate::convert::array::data_chunk_adaptor::{
     DataChunkHandleSlice, NamedDataChunk, SizedFlatVector,
 };
+use crate::convert::scalar::ToDuckDBScalar;
 
 pub trait ToDuckDB {
     fn to_duckdb(&self, chunk: &mut dyn WritableVector) -> VortexResult<()>;
 }
 
+pub fn to_duckdb(array: ArrayRef, chunk: &mut dyn WritableVector) -> VortexResult<()> {
+    if let Some(const_) = array.as_constant() {
+        let value = const_.to_duckdb_scalar();
+        chunk.flat_vector().assign_to_constant(&value);
+        Ok(())
+    } else if array.is_encoding(DictEncoding.id()) {
+        array
+            .as_any()
+            .downcast_ref::<DictArray>()
+            .vortex_expect("dict id checked")
```

Comment:
How about adding explicit returns for all cases?

Reply:
I could, but it this case is an invalid data structure, so I would want to panic here anyway.